packages/peregrine/lib/store/actions/cart/asyncActions.js

```diff
@@ -43,7 +43,10 @@ export const createCart = payload =>
 
             dispatch(actions.getCart.receive(receivePayload));
         } catch (error) {
+            // If we are unable to create a cart, the cart can't function, so
+            // we forcibly throw so the upstream actions won't retry.
             dispatch(actions.getCart.receive(error));
+            throw new Error('Unable to create cart');
```

Comment:
Is there a minimum number of retries we could try here?

Looks like some other actions make that decision based on `error.networkError` - we could retry a few times if it's a network error or something?

Reply:
I want to save retry logic for the apollo link rather than adding it to the actions that we know we are eventually going to remove. But more pertinently, as far as I know there are no possible errors from this mutation aside from server or network errors, which indicate a deeper issue. If you have a working network connection and the server is functioning, calling `createCart` will result in a new cart id or the users cart id if provided a bearer token.